src/main/java/com/product/review/controller/ProductReviewController.java

```
@@ -0,0 +1,69 @@
+
+import com.product.review.domain.Review;
+import com.product.review.exception.OperationNotPerformed;
+import com.product.review.exception.ReviewNotFoundException;
+import com.product.review.service.ProdcutReviewService;
+
+@RestController
+@RequestMapping
+public class ProductReviewController {
+
+	@Autowired
+	private ProdcutReviewService reviewService;
+
+	@GetMapping(value = "/{productid}/reviews")
+	public ResponseEntity<?> getProductReviews(@PathVariable("productid") Integer productID) {
+		List<Review> reviews = reviewService.getProductReviews(productID);
+		return new ResponseEntity<>(reviews, HttpStatus.OK);
+	}
+
+	@GetMapping(value = "/{productid}/reviews/{reviewid}")
+	public ResponseEntity<?> getProductReviews(@PathVariable("productid") Integer productID,
+			@PathVariable("reviewid") Integer reviewID) {
+		Review review = reviewService.getReviewById(productID, reviewID);
+		return new ResponseEntity<>(review, HttpStatus.OK);
+	}
+
+	@PostMapping(value = "/{productid}/reviews")
+	public ResponseEntity<?> createProductReview(@PathVariable("productid") Integer productID,
+			@RequestBody Review review) {
+		review.setProductId(productID);
+		Review review1 = reviewService.saveProductReview(review);
+
+		return new ResponseEntity<>(review1, HttpStatus.CREATED);
+	}
+
+	@PutMapping(value = "/{productid}/reviews/{reviewid}")
+	public ResponseEntity<?> updateProductReview(@PathVariable("productid") Integer productID,
+			@PathVariable("reviewid") Integer reviewID, @RequestBody Review review) {
+		reviewService.getReviewById(reviewID);
+		Review reviewResponse = reviewService.saveProductReview(review);
+		return new ResponseEntity<>(reviewResponse, HttpStatus.OK);
+	}
+
+	@DeleteMapping(value = "/{productid}/reviews/{reviewid}")
+	public ResponseEntity<?> deleteProductReview(@PathVariable("productid") Integer productID,
+			@PathVariable("reviewid") Integer reviewID) {
+		reviewService.getReviewById(reviewID);
```

Do we need to call this here? your service should call getReviewById method and throw ReviewNotFoundException if it is not found else delete the review object